triage/2025-05-04.md

```diff
@@ -0,0 +1,153 @@
+| All ❌✅ (primary)                 | -0.2% | [-0.4%, 0.2%]  | 11    |
+
+> The improvements on primary benchmarks were deemed to outweigh the regressions on secondary benchmarks.
+
+shared-generics: Do not share instantiations that contain local-only types [#138522](https://github.com/rust-lang/rust/pull/138522) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=6e23095adf9209614a45f7f75fea36dad7b92afb&end=0e517d38ad0e72f93c734b14fabd4bb9b7441de6&stat=instructions:u)
+
+| (instructions:u)                   | mean  | range          | count |
+|:----------------------------------:|:-----:|:--------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | 0.9%  | [0.4%, 1.5%]   | 3     |
+| Regressions ❌ <br /> (secondary)  | 0.1%  | [0.1%, 0.1%]   | 1     |
+| Improvements ✅ <br /> (primary)   | -0.5% | [-0.7%, -0.3%] | 6     |
+| Improvements ✅ <br /> (secondary) | -1.0% | [-1.0%, -1.0%] | 1     |
+| All ❌✅ (primary)                 | -0.0% | [-0.7%, 1.5%]  | 9     |
+
+> Marking perf as triaged. There's some improvements, some regressions, but the
+> regressions look like potentially bimodality -- and where they're possibly
+> real hard to justify further digging in. The change seems overall a slight
+> improvement on our benchmarks.
+
+Rollup of 9 pull requests [#140596](https://github.com/rust-lang/rust/pull/140596) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=4824c2bb7445cb2478aab0190c268c939d77a0f6&end=2d5ffc513f1c56b7bc95bacb2519705096e8cc2b&stat=instructions:u)
+
+| (instructions:u)                   | mean  | range          | count |
+|:----------------------------------:|:-----:|:--------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
+| Regressions ❌ <br /> (secondary)  | 0.4%  | [0.2%, 0.5%]   | 2     |
+| Improvements ✅ <br /> (primary)   | -1.2% | [-1.2%, -1.2%] | 1     |
+| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
+| All ❌✅ (primary)                 | -1.2% | [-1.2%, -1.2%] | 1     |
+
+Regression triaged. Regressions look like bimodality/noise, not worth further follow up.
+
+Rollup of 7 pull requests [#140633](https://github.com/rust-lang/rust/pull/140633) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=1bea580f364c65bd5f7380a1056e150df7b8a1a6&end=622ac043764d5d4ffff8de8cf86a1cc938a8a71b&stat=instructions:u)
+
+| (instructions:u)                   | mean  | range          | count |
+|:----------------------------------:|:-----:|:--------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | 1.2%  | [0.2%, 13.4%]  | 77    |
+| Regressions ❌ <br /> (secondary)  | 1.9%  | [0.2%, 11.0%]  | 155   |
+| Improvements ✅ <br /> (primary)   | -1.5% | [-1.5%, -1.5%] | 1     |
+| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
+| All ❌✅ (primary)                 | 1.2%  | [-1.5%, 13.4%] | 78    |
+
+Still investigating the root cause as of writing (see comments on the PR).
```

Comment:
```suggestion
Per investigation on the PR, the root cause is [#139675](https://github.com/rust-lang/rust/pull/139675). Per discussion [here](https://github.com/rust-lang/rust/pull/140652) we expect that this will be mostly resolved in our next LLVM bump, but is likely an acceptable regression in the meantime (primarily to very short compiles) for the feature in question.
```